geowave-accumulo/src/main/java/mil/nga/giat/geowave/accumulo/AccumuloDataStore.java

```diff
@@ -663,6 +664,12 @@ public boolean deleteEntry(
 					adapterId);
 		}
 
+		try {
+			//issue; going to call .flush() internally even if success = false;
+			statsCompositionTool.close();
```

Comment:
So StatsCompositionTool  implements AutoCloseable / has a .close() method - which to me means we should call it when the operation is done / the object goes out of scope.

Here (in the DeleteWriter) the logic seems to say that we only want to .flush() on "success";   (and .close() calls .flush()) - I think what we actually want here is a way to remove the statistic from the local queue if the operation isn't a success, no?   Leaving the StatsCompositionTool  unflushed seems like a chancy practice - depending on if it's used elsewhere it might get flushed regardless.

(didn't implement the delete/removal part, wanted to toss this out for discussion first)